app/models/qualification.rb

```diff
@@ -1,17 +1,22 @@
 class Qualification < ApplicationRecord
+  include ActionView::Helpers::SanitizeHelper
+
   belongs_to :job_application
 
+  SECONDARY_QUALIFICATIONS = %w[gcse as_level a_level other_secondary].freeze
+
   enum category: { gcse: 0, as_level: 1, a_level: 2, other_secondary: 3, undergraduate: 4, postgraduate: 5, other: 6 }
 
-  before_validation :remove_irrelevant_data
+  before_validation :remove_inapplicable_data
 
   def name
     return read_attribute(:name) if read_attribute(:name).present? || other? || other_secondary?
 
     I18n.t("helpers.label.jobseekers_job_application_details_qualifications_category_form.category_options.#{category}")
   end
 
-  def remove_irrelevant_data
+  def remove_inapplicable_data
+    # When `finished_studying` changes, remove answers questions that no longer apply, so that they aren't displayed
     return if finished_studying.nil?
 
     if finished_studying?
```

Comment:
I think you can lose the `self` references here and do `finished_studying_details = ""` etc instead

Reply:
It appears the `self` is required